runtime-sdk/src/callformat.rs

```diff
@@ -196,42 +196,67 @@ pub fn encode_result<C: Context>(
     ctx: &C,
     result: module::CallResult,
     metadata: Metadata,
+) -> CallResult {
+    encode_result_ex(ctx, result, metadata, false /* show_raw_errors */)
+}
+
+pub fn encode_result_ex<C: Context>(
+    ctx: &C,
+    result: module::CallResult,
+    metadata: Metadata,
+    show_raw_errors: bool,
 ) -> CallResult {
     match metadata {
```

Comment:
if you don't want to use a separate function called `encrypt_whatever`, you can do something like this

```rust
if expose_failure && !result.is_success() {
    return result;
}
```
and then you're free of having to check elsewhere

Reply:
The explicit variable usage is nice.

Reply:
Hmm, I think I like having the more explicit flow surrounding `expose_failure`.